packages/treeview/src/useTreeview.ts

```diff
@@ -0,0 +1,211 @@
+ * - collapsible
+ * - a11y controls
+ * @param expandedNodes
+ * @param onChange
+ */
+export function useTreeview<Item = any>({
+  openNodes,
+  onChange = () => undefined,
+  ...options
+}: IUseTreeviewProps<Item> = {}): IUseTreeviewReturnValue<Item> {
+  const [ownFocusedItem, setFocusedItem] = useState<Item>();
+  const { selectedItem, focusedItem, getContainerProps, getItemProps } = useSelection<Item>({
+    direction: 'vertical', // TODO: implement direction
+    defaultSelectedIndex: 0,
+    focusedItem: ownFocusedItem,
+    ...options
+  });
+  const [openNodesState, setOpenNodes] = useState<Item[]>([]);
+  const controlledOpenedState = getControlledValue<Item[] | undefined>(
+    openNodes,
+    Array.from(openNodesState)
+  )!;
+
+  const isControlled = openNodes !== null && openNodes !== undefined;
+  const isNodeExpanded = (item: Item | undefined): boolean => {
+    if (!item) {
+      return false;
+    }
+
+    return controlledOpenedState.includes(item);
+  };
+
+  const toggleParent = (item: Item) => {
+    const newValue = controlledOpenedState.includes(item)
+      ? controlledOpenedState.filter(i => i !== item)
+      : [...controlledOpenedState, item];
+
+    onChange(newValue);
+    if (!isControlled) {
+      setOpenNodes(newValue);
+    }
+  };
+
+  const getTreeProps = ({ label, labelledBy, role = 'tree', ...props }: IGetTreeProps = {}) => {
+    // TODO: Throw error if no label or labelledBy?
+    // TODO: Throw error role is not tree?
+
+    return {
+      role: role === null || role === undefined ? role : 'tree',
+      'aria-label': label,
+      'aria-labelledby': labelledBy,
+      'data-garden-container-id': 'containers.treeview',
+      'data-garden-container-version': PACKAGE_VERSION,
+      ...props
+    };
+  };
+
+  const getTreeItemProps = (
+    {
+      item,
+      role = 'treeitem',
+      nodeType = 'end',
+      onClick,
+      onFocus,
+      onKeyDown,
+      focusRef,
+      ...props
+    }: IGetTreeItemProps<Item> = {} as any
+  ) => {
+    requiredArguments(item, 'item', 'getTreeItemProps');
+    // TODO: Throw error role is not treeitem?
+
+    const expanded = nodeType === 'parent' ? isNodeExpanded(item) : undefined;
+
+    return {
+      role: role === null || role === undefined ? role : 'treeitem',
+      item,
+      focusRef,
+      'aria-expanded': expanded,
+      tabIndex: ownFocusedItem === item ? 0 : -1,
+      'data-garden-container-version': PACKAGE_VERSION,
+      onClick: composeEventHandlers(onClick, (e: MouseEvent) => {
+        if (nodeType !== 'parent') {
+          return;
+        }
+        e.preventDefault();
+        toggleParent(item!);
+      }),
+      onFocus: composeEventHandlers(onFocus, (event: Event) => {
+        event.stopPropagation();
+        setFocusedItem(item);
+      }),
+      onKeyDown: composeEventHandlers(onKeyDown, (event: KeyboardEvent): void => {
+        const target = focusRef.current as HTMLElement;
+
+        event.stopPropagation();
```

Comment:
**issue(blocking):** Don't triggers this for unhandled keys.

Reply:
Also don't trigger if there is modifiers pressed.

Reply:
Good point! Thanks